### ISSUE TITLE
Segfault with textured FBX on 9.0.1

### ISSUE THREAD

**Issue by mwestphal** [f3d-app/f3d#309]
**Describe the bug**
F3D (and VTK) segfaults on a FBX containing textures

**To Reproduce**
 1. Open the file using `./bin/f3d --dry-run /home/glow/f3d/src/testing/data/robot_kyle_walking.fbx`
 2. segfault
 
**Expected behavior**
No segfault

**System Information:**
 - OS: Linux
 - GPU and GPU driver: NVidia

**F3D Information**
Paste the content of `f3d --version`:
```
[glow@arch ~/f3d/build]$ ./bin/f3d --version
f3d 1.2.1

F3D - A fast and minimalist 3D viewer
Version: 1.2.1.
Build date: 2022-06-01 04:40:19.
Build system: Linux x86_64.
Compiler: GNU 12.1.0.
Raytracing module: OFF.
Exodus module: ON.
OpenCASCADE module: OFF.
Assimp module: 5.2.0.
Alembic module: 1.8.3.
VTK version: 9.0.1 (build 1).
Copyright (C) 2019-2021 Kitware SAS.
Copyright (C) 2021-2022 Michael Migliore, Mathieu Westphal.
License BSD-3-Clause.
By Michael Migliore, Mathieu Westphal and Joachim Pouderoux.
```

**Additional context**
gdb stacktrace:

```
Program received signal SIGSEGV, Segmentation fault.
0x00007ffff66caf87 in vtkOpenGLPolyDataMapper::UpdateShaders (this=0x555556009f10, cellBO=..., ren=0x555555c6dc30, actor=0x55555600d480) at /home/glow/work/vtk/others/vtk9.0.1/src/Rendering/OpenGL2/vtkOpenGLPolyDataMapper.cxx:2190
2190        if (newShader != cellBO.Program || cellBO.Program->GetMTime() > cellBO.AttributeUpdateTime)
(gdb) bt
#0  0x00007ffff66caf87 in vtkOpenGLPolyDataMapper::UpdateShaders (this=0x555556009f10, cellBO=..., ren=0x555555c6dc30, actor=0x55555600d480)
    at /home/glow/work/vtk/others/vtk9.0.1/src/Rendering/OpenGL2/vtkOpenGLPolyDataMapper.cxx:2190
#1  0x00007ffff66cf21d in vtkOpenGLPolyDataMapper::RenderPieceDraw (this=0x555556009f10, ren=0x555555c6dc30, actor=0x55555600d480) at /home/glow/work/vtk/others/vtk9.0.1/src/Rendering/OpenGL2/vtkOpenGLPolyDataMapper.cxx:2927
#2  0x00007ffff66cf83c in vtkOpenGLPolyDataMapper::RenderPiece (this=0x555556009f10, ren=0x555555c6dc30, actor=0x55555600d480) at /home/glow/work/vtk/others/vtk9.0.1/src/Rendering/OpenGL2/vtkOpenGLPolyDataMapper.cxx:3031
#3  0x00007ffff5c13eed in vtkPolyDataMapper::Render (this=0x555556009f10, ren=0x555555c6dc30, act=0x55555600d480) at /home/glow/work/vtk/others/vtk9.0.1/src/Rendering/Core/vtkPolyDataMapper.cxx:65
#4  0x00007ffff663639b in vtkOpenGLActor::Render (this=0x55555600d480, ren=0x555555c6dc30, mapper=0x555556009f10) at /home/glow/work/vtk/others/vtk9.0.1/src/Rendering/OpenGL2/vtkOpenGLActor.cxx:104
#5  0x00007ffff5aec20c in vtkActor::RenderOpaqueGeometry (this=0x55555600d480, vp=0x555555c6dc30) at /home/glow/work/vtk/others/vtk9.0.1/src/Rendering/Core/vtkActor.cxx:229
#6  0x00007ffff5c19341 in vtkProp::RenderFilteredOpaqueGeometry (this=0x55555600d480, v=0x555555c6dc30, requiredKeys=0x0) at /home/glow/work/vtk/others/vtk9.0.1/src/Rendering/Core/vtkProp.cxx:273
#7  0x00007ffff66096fd in vtkDefaultPass::RenderFilteredOpaqueGeometry (this=0x555555fdbd90, s=0x7fffffffd2d0) at /home/glow/work/vtk/others/vtk9.0.1/src/Rendering/OpenGL2/vtkDefaultPass.cxx:85
#8  0x00007ffff6635ded in vtkOpaquePass::Render (this=0x555555fdbd90, s=0x7fffffffd2d0) at /home/glow/work/vtk/others/vtk9.0.1/src/Rendering/OpenGL2/vtkOpaquePass.cxx:43
#9  0x00007ffff679d76b in vtkSequencePass::Render (this=0x555555fd9780, s=0x7fffffffd2d0) at /home/glow/work/vtk/others/vtk9.0.1/src/Rendering/OpenGL2/vtkSequencePass.cxx:70
#10 0x00007ffff65e0945 in vtkCameraPass::Render (this=0x555555fda3e0, s=0x7fffffffd2d0) at /home/glow/work/vtk/others/vtk9.0.1/src/Rendering/OpenGL2/vtkCameraPass.cxx:144
#11 0x00007ffff660a512 in vtkDepthImageProcessingPass::RenderDelegate (this=0x555556012040, s=0x7fffffffd690, width=1404, height=1410, newWidth=1404, newHeight=1410, fbo=0x55555603b5d0, colortarget=0x555555ffed30, 
    depthtarget=0x555555ff26d0) at /home/glow/work/vtk/others/vtk9.0.1/src/Rendering/OpenGL2/vtkDepthImageProcessingPass.cxx:130
#12 0x00007ffff662459b in vtkFramebufferPass::Render (this=0x555556012040, s=0x7fffffffd690) at /home/glow/work/vtk/others/vtk9.0.1/src/Rendering/OpenGL2/vtkFramebufferPass.cxx:144
#13 0x00007ffff7f98773 in vtkF3DRenderPass::Render(vtkRenderState const*) () from /home/glow/f3d/build/lib/libf3d.so
#14 0x00007ffff670621e in vtkOpenGLRenderer::DeviceRender (this=0x555555c6dc30) at /home/glow/work/vtk/others/vtk9.0.1/src/Rendering/OpenGL2/vtkOpenGLRenderer.cxx:244
#15 0x00007ffff5c53a36 in vtkRenderer::Render (this=0x555555c6dc30) at /home/glow/work/vtk/others/vtk9.0.1/src/Rendering/Core/vtkRenderer.cxx:377
#16 0x00007ffff5c69dea in vtkRendererCollection::Render (this=0x555555660d90) at /home/glow/work/vtk/others/vtk9.0.1/src/Rendering/Core/vtkRendererCollection.cxx:51
#17 0x00007ffff5c35656 in vtkRenderWindow::DoStereoRender (this=0x55555565bbc0) at /home/glow/work/vtk/others/vtk9.0.1/src/Rendering/Core/vtkRenderWindow.cxx:337
#18 0x00007ffff5c3542f in vtkRenderWindow::Render (this=0x55555565bbc0) at /home/glow/work/vtk/others/vtk9.0.1/src/Rendering/Core/vtkRenderWindow.cxx:297
#19 0x00007ffff6703271 in vtkOpenGLRenderWindow::Render (this=0x55555565bbc0) at /home/glow/work/vtk/others/vtk9.0.1/src/Rendering/OpenGL2/vtkOpenGLRenderWindow.cxx:2042
#20 0x00007ffff67e88c4 in vtkXOpenGLRenderWindow::Render (this=0x55555565bbc0) at /home/glow/work/vtk/others/vtk9.0.1/src/Rendering/OpenGL2/vtkXOpenGLRenderWindow.cxx:1519
#21 0x00007ffff5c3ea50 in vtkRenderWindowInteractor::Render (this=0x5555556520b0) at /home/glow/work/vtk/others/vtk9.0.1/src/Rendering/Core/vtkRenderWindowInteractor.cxx:194
#22 0x00007ffff6541f5a in vtkXRenderWindowInteractor::DispatchEvent (this=0x5555556520b0, event=0x7fffffffe3a0) at /home/glow/work/vtk/others/vtk9.0.1/src/Rendering/UI/vtkXRenderWindowInteractor.cxx:460
#23 0x00007ffff654116a in vtkXRenderWindowInteractor::StartEventLoop (this=0x5555556520b0) at /home/glow/work/vtk/others/vtk9.0.1/src/Rendering/UI/vtkXRenderWindowInteractor.cxx:256
#24 0x00007ffff5c3ebdf in vtkRenderWindowInteractor::Start (this=0x5555556520b0) at /home/glow/work/vtk/others/vtk9.0.1/src/Rendering/Core/vtkRenderWindowInteractor.cxx:242
#25 0x000055555556e711 in F3DStarter::Start(int, char**) ()
#26 0x000055555556c7f5 in main ()
```

cellBO.Program being nullptr causes the segfault. Maybe a version check should be added before adding the texture somewhere.

**Comment by mwestphal** [comment]
same result with slime.fbx

**Comment by mwestphal** [comment]
Actually, this seems to be taken into account in the testing already:

```
if(VTK_VERSION VERSION_GREATER 9.0.20210728) # for TGA support and embedded textures
```

but not in the code.

**Comment by mwestphal** [comment]
Fixed in #316
